watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -400,17 +400,6 @@ def build(self):
         )
 
         if self.config.diffus_calculation == DiffusivityCalculation.none:
-            # Retrieve component string names from user-provided diffusivity_data configuration
-            diffus_data_indices = [i[1] for i in self.config.diffusivity_data.keys()]
-            # identify component string names that were not specified in diffusivity_data
-            missing_diffus_ind = [
-                i for i in self.solute_set if i not in diffus_data_indices
-            ]
-            # warning for components with no diffusivity_data entry
-            if not missing_diffus_ind == []:
-                _log.warning(
-                    f"Diffusivity data was not provided for {missing_diffus_ind}. "
-                )
 
```

Comment:
The original warnings were for mistaken default values being taken when a user neglect to provide correct diffusivity data.   I understand you are trying to get rid of "unnecessary" warnings when diffusivity is not needed. I am not sure if it worths the risk of not having it though.   Without testing it out,  will `diffus_phase_comp` be constructed for all members in `solute_list` including those without `diffusivity_data`?  it looks so to me because the `diffus_p_c` is indexed by `solute_set` and if that is the case, my recommendation is keeping the warning.  I'd leave the decision for you or others to make.

Reply:
The warning is kept exactly in the same implementation, just moved to the method `_diffus_phase_comp`. This ensures the warning only logs when the `diffus_phase_comp` variables are constructed (on demand). It corrected the issue where the warning was stating that diffusivity data was not provided for a flowsheet that the `diffus_phase_comp` is never constructed or used.